### PR TITLE
refactor(bluetooth): removes direct ui invocation from bluetooth thread

### LINKED ISSUE #20429: Remove use of tinyfiledialogs from bluetooth code
Right now the engine is responsible for making UI appear to allow the user to choose an input file: https://github.com/servo/servo/blob/f467bdce1ba95e950b01f59ba284873137bca5d5/components/bluetooth/lib.rs#L369-L397

Instead, the bluetooth code should send a message to the constellation, which can talk to the embedder and receive the final selection to return to the bluetooth code. This means the bluetooth thread needs to have access to a channel which can communicate with the constellation; this can come from [BluetoothManager::new](https://github.com/servo/servo/blob/f467bdce1ba95e950b01f59ba284873137bca5d5/components/bluetooth/lib.rs#L212)  via [BluetoothThreadFactory](https://github.com/servo/servo/blob/f467bdce1ba95e950b01f59ba284873137bca5d5/components/bluetooth/lib.rs#L72-L76). [This](https://github.com/servo/servo/blob/f467bdce1ba95e950b01f59ba284873137bca5d5/components/constellation/constellation.rs#L1845-L1851) is a good model for communicating with the embedder from the constellation.

It's possible to test these changes if you're on linux and you enable the bluetooth preferences and try one of the demo WebBluetooth, but it's also possible to make these changes without testing them.

### PR DESCRIPTION
**Pull request by kwonoj**
Relates to #20429.

This PR is mostly blocked by https://github.com/servo/servo/pull/20480, cause both are nearly identical change in crux and based on review of prior changes this PR need to be updated. Just didn't want assigned issue hanging without visible progresses, so prepped PR to followup further.

(mostly copy-paste from pair PR)

Probably point of review / need to be updated are

1. Communication between components 
Currently it's wired as like below:
```
+----------------+                    +---------------+              +---------+
|                |  constellationMsg  |               | embedderMsg  |         |
| bluetooth_thread +----------------->+ constellation +------------->+ browser |
|                |                    |               |              |         |
+-----+----------+                    +---------------+              +----+----+
      ^                                                                   |
      +-------------------------------------------------------------------+
                                device: String
```
- is this feasible approach?
- does organization of message / fn (where to put consteallation / embedder msg & fns) are legit?

2. Invoking tfd in a separate thread
- This was mainly to align behavior to previous implentation, where tfd was invoked inside of filemanager_thread so does not block main. It may possibly just let block instead.

and of course, a lot of other codes need to follow better practices.

---

- [x] `./mach build -d` does not report any errors
- [ ] `./mach build-geckolib` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #20429 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20578)

### CODE REVIEW

**Comment by paulrouget** [comment]
I'll look at this after we agreed on what needs to be done with PR #20480.

File: components/bluetooth/lib.rs [hunk 1]
```diff
@@ -366,44 +365,32 @@ impl BluetoothManager {
...
-    #[cfg(target_os = "linux")]
     fn select_device(&mut self, devices: Vec<BluetoothDevice>, adapter: &BluetoothAdapter) -> Option<String> {
         if is_mock_adapter(adapter) || opts::get().headless {
-            for device in devices {
+            for device in &devices {
                 if let Ok(address) = device.get_address() {
                     return Some(address);
                 }
             }
-            return None;
```
Comment (paulrouget) [inline components/bluetooth/lib.rs]: why?

**Comment by paulrouget** [comment]
@bors-servo delegate+

Please juste explain why we don't bail early if we use a mock.

File: components/bluetooth/lib.rs (hunk 1 above)
Comment (kwonoj) [reply components/bluetooth/lib.rs]: Thanks for catching, this looks mistake I made while integrate flows of `select_device` function in bluethooth thread. Updated PR to revert changes.

**Comment by kwonoj** [comment]
> Please juste explain why we don't bail early if we use a mock.

Left explanation https://github.com/servo/servo/pull/20578#discussion_r182324131, basically accidental changes while refactoring flows. PR's updated, will try to check in changes once CI says green.

**Comment by kwonoj** [comment]
> command timed out: 1200 seconds without output running ['git', 'fetch', '-t', 'https://github.com/servo/servo', 'auto'], attempting to kill
process killed by signal 9

I tried a couple of times and `mac-rel-wpt3` keep failing with fetching repo, not sure what could cause the same failure in a row.

**Comment by paulrouget** [comment]
Not the only one: http://build.servo.org/builders/mac-rel-wpt3

**Pull request merged**